src/Vulkan/RHI/VkCommandList.cpp

```diff
@@ -244,6 +309,10 @@ ::vk::ImageMemoryBarrier2 GetMemoryBarrierFrom(VkTexture& texture, RHITextureSta
         barrier.srcAccessMask = AccessFlagBits2::eMemoryRead | AccessFlagBits2::eMemoryWrite;
         barrier.srcStageMask = PipelineStageFlagBits2::eAllCommands;
         break;
+    case ImageLayout::eColorAttachmentOptimal:
+        barrier.dstAccessMask = AccessFlagBits2::eShaderWrite;
+        barrier.dstStageMask = PipelineStageFlagBits2::eAllGraphics | PipelineStageFlagBits2::eComputeShader;
+        break;
     default:
         VEX_ASSERT(false, "Transition source image layout not supported");
     }
```

Comment:
In this function I've noticed that the prevLayout sometimes changes dstAccessMask and dstStageMask instead of the src**, is this intended? I've fixed it in my synchronization/constants branch since it was causing issues with incorrect layouts.